### PR TITLE
tests: drivers: dma test on stm32f7 requires nocache memory

### PR DESCRIPTION
**Pull request by FRASTM**
Following the PR https://github.com/zephyrproject-rtos/zephyr/pull/65234
testcase tests/drivers/dma running on sttm32F7 did fail due to ram memory not correclty configured 
CONFIG_NOCACHE_MEMORY is no more used but DCACHE is now enabled for this sstm32f7 soc.

This PR is configuring the SRAM to be non-cachable to PASS the DMA testcases _chan_blen_transfer_ and _loop_transfer_
on the stm32f746zg and stm32f767zi  nucleo boards.

Running the tests/drivers/dma/chan_blen_transfer/ on stm32f7 board (typ. nucleo_f746zg)
```

----- TESTSUITE SUMMARY START ------                                           
                                                                                
SUITE PASS - 100.00% [dma_m2m]: pass = 4, fail = 0, skip = 0, total = 4 duration = 8.056 seconds      
 - PASS - [dma_m2m.test_test_dma0_m2m_chan0_burst16] duration = 2.014 seconds                         
 - PASS - [dma_m2m.test_test_dma0_m2m_chan0_burst8] duration = 2.014 seconds                          
 - PASS - [dma_m2m.test_test_dma0_m2m_chan1_burst16] duration = 2.014 seconds                         
 - PASS - [dma_m2m.test_test_dma0_m2m_chan1_burst8] duration = 2.014 seconds                          
                                                                                
------ TESTSUITE SUMMARY END ------                                          
                                                                                
===================================================================             
PROJECT EXECUTION SUCCESSFUL         
```  

Running the tests/drivers/dma/loop_transfer/ on stm32f7 board (typ. nucleo_f746zg)                                                         
                                                                                                      
```
------ TESTSUITE SUMMARY START ------                                                                 
                                                                                                      
SUITE PASS - 100.00% [dma_m2m_loop]: pass = 3, fail = 0, skip = 0, total = 3 duration = 0.581 seconds 
 - PASS - [dma_m2m_loop.test_test_dma0_m2m_loop] duration = 0.282 seconds                             
 - PASS - [dma_m2m_loop.test_test_dma0_m2m_loop_repeated_start_stop] duration = 0.280 seconds         
 - PASS - [dma_m2m_loop.test_test_dma0_m2m_loop_suspend_resume] duration = 0.019 seconds              
                                                                                                      
------ TESTSUITE SUMMARY END ------                                                                   
                                                                                                      
===================================================================                                   
PROJECT EXECUTION SUCCESSFUL    
```

### CODE REVIEW

**Comment by FRASTM** [comment]
add the testcase for nucleo_f767zi target board

File: tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.conf [hunk 1]
```diff
@@ -1,2 +1 @@
-CONFIG_NOCACHE_MEMORY=y
```
Comment (erwango) [inline tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.conf:1]: Shouldn't we keep this ?

File: tests/drivers/dma/loop_transfer/boards/nucleo_f746zg.overlay [hunk 2]
```diff
@@ -5,3 +5,8 @@
  */
 
 test_dma0: &dma2 { };
+
+/* The test driver expects the SRAM0 region to be non-cachable */
+&sram0 {
+		zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM_NOCACHE) )>;
```
Comment (erwango) [inline tests/drivers/dma/loop_transfer/boards/nucleo_f746zg.overlay:11]: Nitpick, fix indentation:
```c
&sram0 {
	zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM_NOCACHE) )>;
```

File: tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.overlay [hunk 3]
```diff
@@ -11,3 +11,8 @@
 test_dma0: &dma2 {
 	status = "okay";
 };
+
+/* The test driver expects the SRAM0 region to be non-cachable */
+&sram0 {
+		zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM_NOCACHE) )>;
```
Comment (erwango) [inline tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.overlay:17]: ditto

File: tests/drivers/dma/chan_blen_transfer/boards/nucleo_f746zg.conf [hunk 4]
```diff
@@ -1,3 +1,2 @@
 CONFIG_DMA_TRANSFER_CHANNEL_NR_0=0
 CONFIG_DMA_TRANSFER_CHANNEL_NR_1=1
-CONFIG_NOCACHE_MEMORY=y
```
Comment (erwango) [inline tests/drivers/dma/chan_blen_transfer/boards/nucleo_f746zg.conf:3]: On H7, it was required to do:
```
 # Required by BDMA which only has access to do:
# a NOCACHE SRAM4 section. All other DMAs also
# has access to this section.
CONFIG_CODE_DATA_RELOCATION=y
CONFIG_DMA_LOOP_TRANSFER_RELOCATE_SECTION="SRAM4"
```

**Review by erwango — CHANGES_REQUESTED** [review]

File: tests/drivers/dma/chan_blen_transfer/boards/nucleo_f746zg.conf (hunk 4 above)
Comment (FRASTM) [reply tests/drivers/dma/chan_blen_transfer/boards/nucleo_f746zg.conf:3]: due to the BDMA using the SRAM4, this section has to be declared as NOCACHE
For testing with stm32F7, the overlay is setting all the sram0 as NOCACHE

**Review by erwango — APPROVED** [review]

File: tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.overlay (hunk 3 above)
Comment (FRASTM) [reply tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.overlay]: done

File: tests/drivers/dma/loop_transfer/boards/nucleo_f746zg.overlay (hunk 2 above)
Comment (FRASTM) [reply tests/drivers/dma/loop_transfer/boards/nucleo_f746zg.overlay]: done

File: tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.conf (hunk 1 above)
Comment (FRASTM) [reply tests/drivers/dma/loop_transfer/boards/nucleo_f767zi.conf:1]: useless

**Review by teburd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**